.murdock

```diff
@@ -240,37 +239,43 @@ compile() {
     [ $# -ne 2 ] && error "$0: compile: invalid parameters (expected \$appdir \$board:\$toolchain)"
     [ ! -d "$appdir" ] && error "$0: compile: error: application directory \"$appdir\" doesn't exist"
 
-    # compile
+    # We compile a first time with Kconfig based dependency
+    # resolution for regression purposes. $TEST_KCONFIG contains a
+    # list of board-application tuples that are currently modeled to
+    # run with Kconfig
+
+    should_check_kconfig_hash=0
+
+    for app in $(get_kconfig_test_apps "${board}")
+    do
+        if [ "${appdir}" = "${app}" ]; then
+            should_check_kconfig_hash=1
+            BOARD=$board make -C${appdir} clean
+            CCACHE_BASEDIR="$(pwd)" BOARD=$board TOOLCHAIN=$toolchain RIOT_CI_BUILD=1 TEST_KCONFIG=1 \
+                            make -C${appdir} all test-input-hash -j${JOBS:-4}
+            RES=$?
+            if [ $RES -eq 0 ]; then
+                kconfig_test_hash=$(test_hash_calc "$BINDIR")
```

Comment:
```suggestion
                kconfig_test_hash=$(test_hash_calc "${BINDIR}")
```

Reply:
Seems like shellcheck doesn't like a big part of this script though:

<details><summary><b>shellcheck .murdock</b></summary>

```bash

In .murdock line 3:
: ${TEST_BOARDS_AVAILABLE:="esp32-wroom-32 samr21-xpro"}
  ^-- SC2223: This default assignment may cause DoS due to globbing. Quote it.


In .murdock line 4:
: ${TEST_BOARDS_LLVM_COMPILE:="iotlab-m3 native nrf52dk mulle nucleo-f401re samr21-xpro slstk3402a"}
  ^-- SC2223: This default assignment may cause DoS due to globbing. Quote it.


In .murdock line 6:
: ${TEST_KCONFIG_samr21_xpro:="examples/hello-world tests/periph_*"}
  ^-- SC2223: This default assignment may cause DoS due to globbing. Quote it.


In .murdock line 45:
    local label="${1}"
    ^---------^ SC2039: In POSIX sh, 'local' is undefined.


In .murdock line 82:
            error "$0: hook \"${MURDOCK_HOOK} $@\" failed!"
                                              ^-- SC2145: Argument mixes string and array. Use * or separate argument.


In .murdock line 94:
    local needle="$1"
    ^----------^ SC2039: In POSIX sh, 'local' is undefined.


In .murdock line 95:
    local haystack="$2"
    ^------------^ SC2039: In POSIX sh, 'local' is undefined.


In .murdock line 110:
        echo -n "_grep -E ($1"
             ^-- SC2039: In POSIX sh, echo flags are undefined.


In .murdock line 112:
        for i in $*; do
                 ^-- SC2048: Use "$@" (with quotes) to prevent whitespace problems.


In .murdock line 113:
            echo -n "|$i"
                 ^-- SC2039: In POSIX sh, echo flags are undefined.


In .murdock line 122:
        | $(_greplist $APPS) | sort
          ^----------------^ SC2091: Remove surrounding $() to avoid executing output.
                      ^---^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
        | $(_greplist "$APPS") | sort


In .murdock line 128:
    local appdir=$1
    ^----------^ SC2039: In POSIX sh, 'local' is undefined.


In .murdock line 129:
    local boards="$(make --no-print-directory -C$appdir info-boards-supported 2>/dev/null || echo broken)"
    ^----------^ SC2039: In POSIX sh, 'local' is undefined.
          ^----^ SC2155: Declare and assign separately to avoid masking return values.
                                                ^-----^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
    local boards="$(make --no-print-directory -C"$appdir" info-boards-supported 2>/dev/null || echo broken)"


In .murdock line 138:
        echo $board
             ^----^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
        echo "$board"


In .murdock line 139:
    done | $(_greplist $BOARDS)
           ^------------------^ SC2091: Remove surrounding $() to avoid executing output.
                       ^-----^ SC2153: Possible misspelling: BOARDS may not be assigned, but boards is.
                       ^-----^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
    done | $(_greplist "$BOARDS")


In .murdock line 143:
    local appdir=$1
    ^----------^ SC2039: In POSIX sh, 'local' is undefined.


In .murdock line 144:
    local board=$2
    ^---------^ SC2039: In POSIX sh, 'local' is undefined.


In .murdock line 145:
    local toolchains="gnu"
    ^--------------^ SC2039: In POSIX sh, 'local' is undefined.


In .murdock line 148:
        toolchains="$(make -s --no-print-directory -C${appdir} BOARD=${board} \
                                                     ^-------^ SC2086: Double quote to prevent globbing and word splitting.
                                                                     ^------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
        toolchains="$(make -s --no-print-directory -C"${appdir}" BOARD="${board}" \


In .murdock line 158:
    local appdir=$1
    ^----------^ SC2039: In POSIX sh, 'local' is undefined.


In .murdock line 159:
    local boards="$(get_supported_boards $appdir)"
    ^----------^ SC2039: In POSIX sh, 'local' is undefined.
          ^----^ SC2155: Declare and assign separately to avoid masking return values.
                                         ^-----^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
    local boards="$(get_supported_boards "$appdir")"


In .murdock line 163:
    local prefix="$*"
    ^----------^ SC2039: In POSIX sh, 'local' is undefined.


In .murdock line 172:
        for toolchain in $(get_supported_toolchains $appdir $board)
                                                    ^-----^ SC2086: Double quote to prevent globbing and word splitting.
                                                            ^----^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
        for toolchain in $(get_supported_toolchains "$appdir" "$board")


In .murdock line 174:
            echo $prefix $appdir $board:$toolchain
                 ^-----^ SC2086: Double quote to prevent globbing and word splitting.
                         ^-----^ SC2086: Double quote to prevent globbing and word splitting.
                                 ^----^ SC2086: Double quote to prevent globbing and word splitting.
                                        ^--------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
            echo "$prefix" "$appdir" "$board":"$toolchain"


In .murdock line 176:
    done | $(_greplist $BOARDS)
           ^------------------^ SC2091: Remove surrounding $() to avoid executing output.
                       ^-----^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
    done | $(_greplist "$BOARDS")


In .murdock line 183:
        dwqc ${DWQ_ENV} -s \
             ^--------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
        dwqc "${DWQ_ENV}" -s \


In .murdock line 194:
    local bindir=$1
    ^----------^ SC2039: In POSIX sh, 'local' is undefined.


In .murdock line 205:
    cat ${bindir}/test-input-hash.sha1 | cut -f1 -d' ' | sha1sum | cut -f1 -d' '
        ^-------^ SC2086: Double quote to prevent globbing and word splitting.
        ^----------------------------^ SC2002: Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.

Did you mean:
    cat "${bindir}"/test-input-hash.sha1 | cut -f1 -d' ' | sha1sum | cut -f1 -d' '


In .murdock line 210:
    test -n "$(redis-cli get $1)" > /dev/null
                             ^-- SC2086: Double quote to prevent globbing and word splitting.
                                  ^---------^ SC2065: This is interpreted as a shell file redirection, not a comparison.

Did you mean:
    test -n "$(redis-cli get "$1")" > /dev/null


In .murdock line 219:
    local appdir=$1
    ^----------^ SC2039: In POSIX sh, 'local' is undefined.


In .murdock line 220:
    local board=$(echo $2 | cut -f 1 -d':')
    ^---------^ SC2039: In POSIX sh, 'local' is undefined.
          ^---^ SC2155: Declare and assign separately to avoid masking return values.
                       ^-- SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
    local board=$(echo "$2" | cut -f 1 -d':')


In .murdock line 221:
    local toolchain=$(echo $2 | cut -f 2 -d':')
    ^-------------^ SC2039: In POSIX sh, 'local' is undefined.
          ^-------^ SC2155: Declare and assign separately to avoid masking return values.
                           ^-- SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
    local toolchain=$(echo "$2" | cut -f 2 -d':')


In .murdock line 226:
        make -C$appdir info-boards-supported && echo "$0: success. no idea what's wrong." || echo "$0: failed!"
               ^-----^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
        make -C"$appdir" info-boards-supported && echo "$0: success. no idea what's wrong." || echo "$0: failed!"


In .murdock line 231:
    export BINDIR="$(pwd)/build"
           ^----^ SC2155: Declare and assign separately to avoid masking return values.


In .murdock line 235:
    rm -rf ${BINDIR}
           ^-------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
    rm -rf "${BINDIR}"


In .murdock line 245:
        make -C${appdir} clean all test-input-hash -j${JOBS:-4}
               ^-------^ SC2086: Double quote to prevent globbing and word splitting.
                                                     ^--------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
        make -C"${appdir}" clean all test-input-hash -j"${JOBS:-4}"


In .murdock line 256:
        _board=$(echo ${board} | tr '-' '_')
                      ^------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
        _board=$(echo "${board}" | tr '-' '_')


In .murdock line 261:
                BOARD=$board make -C${appdir} clean
                                    ^-------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
                BOARD=$board make -C"${appdir}" clean


In .murdock line 263:
                              make -C${appdir} all test-input-hash -j${JOBS:-4}
                                     ^-------^ SC2086: Double quote to prevent globbing and word splitting.
                                                                     ^--------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
                              make -C"${appdir}" all test-input-hash -j"${JOBS:-4}"


In .murdock line 267:
                    if [ ${new_test_hash} != ${test_hash} ]; then
                         ^--------------^ SC2086: Double quote to prevent globbing and word splitting.
                                             ^----------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
                    if [ "${new_test_hash}" != "${test_hash}" ]; then


In .murdock line 278:
        if [ $RUN_TESTS -eq 1 -o "$board" = "native" ]; then
                              ^-- SC2166: Prefer [ p ] || [ q ] as [ p -o q ] is not well defined.


In .murdock line 281:
                    BOARD=$board make -C${appdir} test
                                        ^-------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
                    BOARD=$board make -C"${appdir}" test


In .murdock line 285:
                    if test_cache_get $test_hash; then
                                      ^--------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
                    if test_cache_get "$test_hash"; then


In .murdock line 289:
                            make -C${appdir} test-murdock
                                   ^-------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
                            make -C"${appdir}" test-murdock


In .murdock line 297:
    if [ -d ${BINDIR} ]
            ^-------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
    if [ -d "${BINDIR}" ]


In .murdock line 299:
        echo "-- build directory size: $(du -sh ${BINDIR} | cut -f1)"
                                                ^-------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
        echo "-- build directory size: $(du -sh "${BINDIR}" | cut -f1)"


In .murdock line 302:
        rm -rf ${BINDIR}
               ^-------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
        rm -rf "${BINDIR}"


In .murdock line 309:
    local appdir=$1
    ^----------^ SC2039: In POSIX sh, 'local' is undefined.


In .murdock line 310:
    local board=$(echo $2 | cut -f 1 -d':')
    ^---------^ SC2039: In POSIX sh, 'local' is undefined.
          ^---^ SC2155: Declare and assign separately to avoid masking return values.
                       ^-- SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
    local board=$(echo "$2" | cut -f 1 -d':')


In .murdock line 311:
    local toolchain=$(echo $2 | cut -f 2 -d':')
    ^-------------^ SC2039: In POSIX sh, 'local' is undefined.
          ^-------^ SC2155: Declare and assign separately to avoid masking return values.
                           ^-- SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
    local toolchain=$(echo "$2" | cut -f 2 -d':')


In .murdock line 317:
    local files=""
    ^---------^ SC2039: In POSIX sh, 'local' is undefined.


In .murdock line 322:
          local relpath="$(realpath --relative-to ${BINDIR} ${filename})"
          ^-----------^ SC2039: In POSIX sh, 'local' is undefined.
                ^-----^ SC2155: Declare and assign separately to avoid masking return values.
                                                  ^-------^ SC2086: Double quote to prevent globbing and word splitting.
                                                            ^---------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
          local relpath="$(realpath --relative-to "${BINDIR}" "${filename}")"


In .murdock line 329:
        local remote_bindir="${appdir}/bin/${board}"
        ^-----------------^ SC2039: In POSIX sh, 'local' is undefined.


In .murdock line 334:
        ${DWQ_ENV} \
        ^--------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
        "${DWQ_ENV}" \


In .murdock line 336:
        --queue ${TEST_QUEUE:-$board} \
                ^-------------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
        --queue "${TEST_QUEUE:-$board}" \


In .murdock line 338:
        $files \
        ^----^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
        "$files" \


In .murdock line 343:
    local appdir=$1
    ^----------^ SC2039: In POSIX sh, 'local' is undefined.


In .murdock line 344:
    local board=$(echo $2 | cut -f 1 -d':')
    ^---------^ SC2039: In POSIX sh, 'local' is undefined.
          ^---^ SC2155: Declare and assign separately to avoid masking return values.
                       ^-- SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
    local board=$(echo "$2" | cut -f 1 -d':')


In .murdock line 345:
    local toolchain=$(echo $2 | cut -f 2 -d':')
    ^-------------^ SC2039: In POSIX sh, 'local' is undefined.
          ^-------^ SC2155: Declare and assign separately to avoid masking return values.
                           ^-- SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
    local toolchain=$(echo "$2" | cut -f 2 -d':')


In .murdock line 351:
    BOARD=$board TOOLCHAIN=${toolchain} make -C$appdir flash-only termdeps -j2
                                               ^-----^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
    BOARD=$board TOOLCHAIN=${toolchain} make -C"$appdir" flash-only termdeps -j2


In .murdock line 354:
    BOARD=$board TOOLCHAIN=${toolchain} make -C$appdir test
                                               ^-----^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
    BOARD=$board TOOLCHAIN=${toolchain} make -C"$appdir" test


In .murdock line 357:
    if [ $RES -eq 0 -a -n "$TEST_HASH" ]; then
                    ^-- SC2166: Prefer [ p ] && [ q ] as [ p -a q ] is not well defined.
                           ^--------^ SC2153: Possible misspelling: TEST_HASH may not be assigned, but test_hash is.


In .murdock line 358:
        echo -n "-- saving test result to cache: "
             ^-- SC2039: In POSIX sh, echo flags are undefined.


In .murdock line 359:
        test_cache_put $TEST_HASH
                       ^--------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
        test_cache_put "$TEST_HASH"


In .murdock line 381:
$*
^-- SC2048: Use "$@" (with quotes) to prevent whitespace problems.

For more information:
  https://www.shellcheck.net/wiki/SC2145 -- Argument mixes string and array. ...
  https://www.shellcheck.net/wiki/SC2039 -- In POSIX sh, 'local' is undefined.
  https://www.shellcheck.net/wiki/SC2048 -- Use "$@" (with quotes) to prevent...

```

</details>

Reply:
I fixed this one too. Do you think we should piggyback all those changes here @fjmolinas ?

Reply:
> I fixed this one too. Do you think we should piggyback all those changes here @fjmolinas ?

No, just pointing it out

Reply:
@bergzand is aware of the shellcheck issues in this script. I think he's willing to provide a PR addressing them.